tests/NSubstitute.Acceptance.Specs/FieldReports/Issue262_NonPublicSetterCall.cs

```diff
@@ -0,0 +1,53 @@
+using NUnit.Framework;
+
+namespace NSubstitute.Acceptance.Specs.FieldReports
+{
+    /// <summary>
+    /// Scenarios for the issue: https://github.com/nsubstitute/NSubstitute/issues/626
+    ///
+    /// Do not test internal members, as we don't want to set InternalsVisibleTo attribute.
+    /// </summary>
+    public class Issue262_NonPublicSetterCall
+    {
+        [Test]
+        public void ShouldHandleProtectedProperties()
+        {
+            var substitute = Substitute.For<TestClass>();
+
+            substitute.ProtectedPropProxy = 42;
+
+            var result = substitute.ProtectedPropProxy;
+            Assert.That(result, Is.EqualTo(expected: 42));
+            substitute.Received().ProtectedPropProxy = Arg.Any<int>();
+        }
+
+        [Test]
+        public void ShouldHandlePropertiesWithDifferentAccessorVisibility()
+        {
+            var subs = Substitute.For<TestClass>();
+
+            subs.PublicGetProtectedSetPropSetter = 42;
+
+            var result = subs.PublicGetProtectedSetProp;
+            Assert.That(result, Is.EqualTo(expected: 42));
+        }
+
+        public abstract class TestClass
+        {
+            protected abstract int ProtectedProp { get; set; }
+
+            public int ProtectedPropProxy
+            {
+                get => ProtectedProp;
+                set => ProtectedProp = value;
+            }
+
+            public abstract int PublicGetProtectedSetProp { get; protected set; }
+
+            public int PublicGetProtectedSetPropSetter
+            {
+                set => PublicGetProtectedSetProp = value;
+            }
```

Comment:
Thanks for the idea of `Prop { protected get; set; }` - added it as well 👍 

```c#
public virtual int OtherProp { private get; set; }
```

As for the property with `private` getter, we cannot override and intercept the getter here. When proxy is created, we intercept a setter only. Therefore, no matter what we assign - it will be never possible to read it.

The same actually happens with the property like this:

```c#
public virtual int OtherProp { get; private set; }
```

Here you can override a getter only.

If you create a substitute, it will be very broken property functionality-wise, as it will "void" all the assigned values.

Notice that the scenarios above are only applicable to the partial mocks, which are not that frequent. For interface mocks which are way more popular those scenarios are not applicable.